samples/subsys/logging/nrf_coresight_stm/src/main.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_frontend_stmesp.h>
+LOG_MODULE_REGISTER(app);
+
+#define TEST_LOG(rpt, item)                                                                        \
+	({                                                                                         \
+		uint32_t t = k_cycle_get_32();                                                     \
+		for (uint32_t i = 0; i < rpt; i++) {                                               \
+			__DEBRACKET item;                                                          \
+		}                                                                                  \
+		t = k_cycle_get_32() - t;                                                          \
+		k_msleep(200);                                                                     \
+		t;                                                                                 \
+	})
+
+static char *core_name = "unknown";
+
+static void get_core_name(void)
+{
+	if (strstr(CONFIG_BOARD_TARGET, "cpuapp")) {
+		core_name = "app";
+	} else if (strstr(CONFIG_BOARD_TARGET, "cpurad")) {
+		core_name = "rad";
+	} else if (strstr(CONFIG_BOARD_TARGET, "cpuppr")) {
+		core_name = "ppr";
+	}
```

Comment:
Is it intentional to use SoC-specific CPU names in a generic subsystem sample?

Reply:
sample is called `nrf_coresight_stm` and it is nordic specific. So it's actually better to move it to `samples/boards/nrf`